bears/vcs/git/GitCommitBear.py

```diff
@@ -203,3 +241,76 @@ def check_body(self, body,
                                'characters.'.format(body_line_length))
+
+    def check_issue(self, body,
+                    body_close_issue: bool=False,
+                    body_close_issue_full_url: bool=False,
+                    body_close_issue_on_last_line: bool=False):
+        """
+        Check for matching issue related references and URLs.
+
+        :param body:
+            Body of the commit message of HEAD.
+        :param body_close_issue:
+            Whether to check for the presence of issue close reference
+            within the commit body by retrieving host information from git
+            configuration. GitHub and GitLab support auto closing issues with
+            commit messages. Checks for matching keywords in the commit body.
+            By default, if none of ``body_close_issue_full_url`` and
+            ``body_close_issue_on_last_line`` are enabled, checks for presence
+            of short references like ``closes #213``. Otherwise behaves
+            according to other chosen flags.
+            More on keywords follows.
+            [GitHub](https://help.github.com/articles/closing-issues-via-commit-messages/)
+            [GitLab](https://docs.gitlab.com/ce/user/project/issues/automatic_issue_closing.html)
+        :param body_close_issue_full_url:
+            Checks the presence of issue close reference with a full URL
+            related to some issue. Works along with ``body_close_issue``.
+        :param body_close_issue_on_last_line:
+            When enabled, checks for issue close reference presence on the
+            last line of the commit body. Works along with
+            ``body_close_issue``.
+        """
+        if not body_close_issue:
+            return
+
+        host = self.get_host_from_remotes()
+        if host not in self.SUPPORTED_HOSTS:
+            return
+
+        if not body_close_issue_on_last_line:
+            result_message = ('No {} reference found in the body at HEAD '
+                              'commit.')
+        else:
+            body = body.splitlines()[-1]
+            result_message = ('No {} reference found in the body at HEAD '
+                              'commit in the last line.')
+
+        keywords = self.SUPPORTED_HOSTS[host]
+        good_ref = False
+        body = body.lower()
+
+        if not any(key in body for key in keywords):
+            yield Result(self, result_message.format('issue'))
+            return
+
+        if not body_close_issue_full_url:
+            if not re.search('#[1-9][0-9]*', body):
+                yield Result(self, result_message.format('short issue'))
+            return
+
+        for match in re.findall(r'https?://\S+', body):
+            netloc, path = urlparse(match)[1:3]
+            if host not in netloc or '/issues/' not in path:
+                continue
+            issue_num = path.split('/')[-1]
+            try:
+                issue_num = int(issue_num)
```

Comment:
I think you mean 313, and also if the `int` conversion fails the result won't be printed there anyway, you still get the string from before.